doc/source/area_def_list.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2022 Satpy developers
+#
+# This file is part of satpy.
+#
+# satpy is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# satpy is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with satpy.  If not, see <http://www.gnu.org/licenses/>.
+"""Module for autogenerating a list and overview of available area definitions ."""
+
+from pyresample.area_config import _read_yaml_area_file_content
+from pyresample.formatting_html import area_repr
+
+from satpy.resample import get_area_def, get_area_file
+
+
+def generate_area_def_list():
+    """Create list of available area definitions with overview plot.
+
+    Returns:
+        str
+    """
+    area_list = []
+
+    template = ("{area_name}\n"
+                "{n:->{header_title_length}}\n\n"
+                ".. raw:: html\n\n"
+                "{content}\n\n"
+                "     <hr>\n\n")
+
+    area_file = get_area_file()[0]
+    for aname in list(_read_yaml_area_file_content(area_file).keys()):
+        area = get_area_def(aname)
+        if hasattr(area, "_repr_html_"):
+            content = "\n".join([x.rjust(len(x) + 5) for x in area_repr(area, include_header=False).split("\n")])
+            area_list.append(template.format(area_name=aname, n="", header_title_length=len(aname),
+                                             content=content))
+        else:
+            pass
```

Comment:
Doesn't this function rather belong to pyresample?

Reply:
I think it doesn't really make sense to include in in pyresample because there is no area definition yaml included and this is only useful to generate a rst with a list of area definitions for the documentation.

Reply:
It is true that pyresample does not have a default yaml file, but it is pyresample that contains all the functions for reading such yaml files. And also, I would argue that for the sake of separation of concerns, it should be included there.
In practice, I wouldn't be against having a default yaml file with a few generic areas in pyresample by the way, that would probably help potential pyresample users in the future...

Reply:
Wouldn't `areas.yaml` fit better with pyresample entirely?  But pyresample does not have a concept of `SATPY_CONFIG_PATH` or a config directory search, so it wouldn't know where to find `areas.yaml` anyway?

Reply:
Not that anyone asked for my opinion, but I agree with wanting this in pyresample. I'm not a big fan of a default set of areas in pyresample except only as a few examples (like global lon/lat, one LCC, one polar-stereographic, etc)...generic areas. Pyresample is the low-level functionality, but it doesn't know anything about satellite-specific areas or that it is used for "science" at all so to me it is a fine line.

Reply:
A while ago, I found a small list of generic areas, like europe, north and south america, etc. I'll try to find it again, so we can use that.

Reply:
As far as I see, those are just area _names_ without defining anything `AreaDefinition`y that we could use.